packages/babel-plugin-proposal-class-properties/src/index.js

```diff
@@ -33,7 +33,7 @@ export default function(api, options) {
     },
 
     ReferencedIdentifier(path) {
-      if (this.classRef === path.scope.getBinding(path.node.name)) {
+      if (this.classBinding === path.scope.getBinding(path.node.name)) {
```

Comment:
Instead of using `false` as a special value to indicate tha the class doen't have an `id`, I think that we should let it be undefined (as it was) and check `this.classBinding && this.classBinding === ...` here.